test_mypy.yml fails hilariously on Python 3.10

The error list is long than my scrollback buffer, but here's an example:

```
/Users/hynek/FOSS/attrs/tests/test_mypy.yml:118:
E   pytest_mypy_plugins.utils.TypecheckAssertionError: Invalid output:
E   Expected:
E     main:6: error: Incompatible return value type (got "int", expected "str") (diff)
E     main:11: error: Incompatible return value type (got "int", expected "str") (diff)
E     main:16: error: Incompatible return value type (got "int", expected "str") (diff)
E     main:21: error: Incompatible return value type (got "int", expected "str") (diff)
E   Actual:
E     ../../../../../../Users/hynek/FOSS/attrs/.tox/py310/lib/python3.10/site-packages/mypy/typeshed/stdlib/3/types.pyi:334: error: Name '_NotImplementedType' is not defined (diff)
E     main:6: error: Incompatible return value type (got "int", expected "str") (diff)
E     main:11: error: Incompatible return value type (got "int", expected "str") (diff)
E     main:16: error: Incompatible return value type (got "int", expected "str") (diff)
E     main:21: error: Incompatible return value type (got "int", expected "str") (diff)
E
E   Alignment of first line difference:
E     E: main:6: error: Incompatible return value type (got "int", expected "str"...
E     A: ../../../../../../Users/hynek/FOSS/attrs/.tox/py310/lib/python3.10/site-...
E        ^
```

or

```
/Users/hynek/FOSS/attrs/tests/test_mypy.yml:277:
E   pytest_mypy_plugins.utils.TypecheckAssertionError: Invalid output:
E   Expected:
E     main:5: note: Revealed type is 'def (a: builtins.int) -> main.A' (diff)
E     main:7: error: Unsupported left operand type for < ("A") (diff)
E     main:8: error: Unsupported left operand type for <= ("A") (diff)
E     main:9: error: Unsupported left operand type for > ("A") (diff)
E     ...
E   Actual:
E     ../../../../../../Users/hynek/FOSS/attrs/.tox/py310/lib/python3.10/site-packages/mypy/typeshed/stdlib/3/types.pyi:334: error: Name '_NotImplementedType' is not defined (diff)
E     main:5: note: Revealed type is 'def (a: builtins.int) -> main.A' (diff)
E     main:7: error: Unsupported left operand type for < ("A") (diff)
E     main:8: error: Unsupported left operand type for <= ("A") (diff)
E     main:9: error: Unsupported left operand type for > ("A") (diff)
E     ...
E
E   Alignment of first line difference:
E     E: main:5: note: Revealed type is 'def (a: builtins.int) -> main.A'...
E     A: ../../../../../../Users/hynek/FOSS/attrs/.tox/py310/lib/python3.10/site-...
E        ^
```

I'm not sure what to do about it at all but I guess we _should_ fix it going forward.

--- a/conftest.py
+++ b/conftest.py
@@ -27,9 +27,3 @@ if sys.version_info[:2] < (3, 6):
     )
 if not PY310:
     collect_ignore.extend(["tests/test_pattern_matching.py"])
-if PY310:
-    collect_ignore.extend(
-        [
-            "tests/test_mypy.yml",
-        ]
-    )

--- a/tox.ini
+++ b/tox.ini
@@ -128,7 +128,7 @@ commands = towncrier --draft
 
 [testenv:typing]
 basepython = python3.8
-deps = mypy>=0.800
+deps = mypy>=0.902
 commands =
     mypy src/attr/__init__.pyi src/attr/_version_info.pyi src/attr/converters.pyi src/attr/exceptions.pyi src/attr/filters.pyi src/attr/setters.pyi src/attr/validators.pyi
     mypy tests/typing_example.py

